src/IIIFPresentation/API/Features/Manifest/ManifestWriteService.cs

```diff
@@ -198,6 +198,18 @@ private async Task<PresUpdateResult> UpdateInternal(UpsertManifestRequest reques
             return ErrorHelper.EtagNonMatching<PresentationManifest>();
         }
 
+        if (!request.PresentationManifest.PaintedResources.HasAsset() &&
```

Comment:
Store result of these in vars, rather than calculating potentially twice?

```cs
var hasAsset = request.PresentationManifest.PaintedResources.HasAsset();
var noBatches = existingManifest.Batches.IsNullOrEmpty();

if (!hasAssets && noBatches) { .. }
if (hasAsset && !noBatches) { .. }
```